packages/venia-concept/src/components/ProductImageCarousel/thumbnail.js

```diff
@@ -1,59 +1,56 @@
-import React, { Component } from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 
 import { resourceUrl } from 'src/drivers';
-import classify from 'src/classify';
+import { mergeClasses } from 'src/classify';
 import defaultClasses from './thumbnail.css';
 import { transparentPlaceholder } from 'src/shared/images';
+import { WindowSizeContext } from 'src/components/App/WindowSizeContext';
 
-class Thumbnail extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string,
-            rootSelected: PropTypes.string
-        }),
-        isActive: PropTypes.bool,
-        item: PropTypes.shape({
-            label: PropTypes.string,
-            file: PropTypes.string.isRequired
-        }),
-        itemIndex: PropTypes.number,
-        onClickHandler: PropTypes.func.isRequired
-    };
-    // TODO: When we implement hooks/mergeClasses from #1078 we can replace this
-    // with a useWindowSize hook and rerender when the width changes across the
-    // breakpoint. This is important when changing from portrait to landscape.
-    isDesktop = () => {
-        return window.innerWidth >= 1024;
-    };
-
-    onClickHandlerWrapper = () => {
-        const { onClickHandler, itemIndex } = this.props;
+function Thumbnail(props) {
+    const { windowSize } = useContext(WindowSizeContext);
```

Comment:
This isn't working. For some reason, `WindowSizeContext` is an object containing the exports from the WindowSizeContext file:

```js
WindowSizeContext === {
  WindowSizeContext: (...),
  WindowSizeContextProvider: (...),
  ...
}
```

What did I do wrong? I exported both as non-defaults in the context file and I'm importing by name here so why is it resolving the exported object?

Reply:
Or maybe it's working properly and I'm just not seeing the bug I wrote...